### PR TITLE
Fix resource listing script

### PR DESCRIPTION
**Pull request by houkms**
This PR fixes the "command not found" issue in pipeline of PR #95.

### CODE REVIEW

**Comment by metacpp** [comment]
@houkms I am curious that Azure pipeline did not fail for #95, I guess it's because of `fail at stderr` is disabled. Anyway, please manually check the log file to make sure every step successfully run.

File: list_resources.sh [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+			# find resource directory recursively
+			findResourceDir $work_dir
+		fi
+	done
 }
-
+# go to magic-module-specs directory
+cd `dirname $0`
+specs_dir='./specs'
 findResourceDir $specs_dir
```
Comment (metacpp) [inline list_resources.sh]: Please add new line

File: list_resources.sh [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-        	if [ -n "$(git status -s $work_dir)" ] && [ -f $work_dir"/api.yaml" ] \
-        		&& ([ -f $work_dir"/ansible.yaml" ] || [ -f $work_dir"/terraform.yaml" ]); then
-				echo ${work_dir:8}		# remove the directory prefix './specs/'
+findResourceDir(){
+	for file in `ls $1`
+	do
+		work_dir=$1"/"$file
+		if [ -d $work_dir ]
+		then
+			if [ -n "$(git status -s $work_dir)" ] && [ -f $work_dir"/api.yaml" ] \
```
Comment (metacpp) [inline list_resources.sh]: We don't need to check existence of YAML files, MM can handle it very well.

To make the code logic simple, please just get a list of changed/added folder names, that's all.

File: list_resources.sh [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
+				&& ([ -f $work_dir"/ansible.yaml" ] || [ -f $work_dir"/terraform.yaml" ]);
+			then
+				# remove the directory prefix './specs'
+				expr substr "$work_dir" 9 "${#work_dir}"
 			fi
-            findResourceDir $work_dir	# find resource directory recursively
-        fi
-    done
+			# find resource directory recursively
+			findResourceDir $work_dir
```
Comment (metacpp) [inline list_resources.sh]: No need to do it recursively, we only try first level under the root folder of specs.

File: list_resources.sh [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
+		fi
+	done
 }
-
+# go to magic-module-specs directory
+cd `dirname $0`
+specs_dir='./specs'
```
Comment (metacpp) [inline list_resources.sh]: Please get absolute path of `specs` then pass it your function. Don't change current directory and use relative directory.

File: list_resources.sh (hunk 2 above)
Comment (houkms) [reply list_resources.sh]: Removed.

File: list_resources.sh (hunk 1 above)
Comment (houkms) [reply list_resources.sh]: Added.

File: list_resources.sh (hunk 3 above)
Comment (houkms) [reply list_resources.sh]: Removed.

File: list_resources.sh (hunk 4 above)
Comment (houkms) [reply list_resources.sh]: Git command needs to work in the repo directory. What's more, it's in a new shell script, and it won't change the work directory of the caller.

**Comment by houkms** [comment]
Codes are modified according to the comments.

**Comment by metacpp** [comment]
@houkms you need to resolve the conflicts.

File: list_resources.sh (hunk 4 above)
Comment (metacpp) [reply list_resources.sh]: Why not directly set the specs_dir to be `dirname $0`/specs ?

Comment (houkms) [reply list_resources.sh]: Because git needs to run in the root directory of magic-modules-spec.

**Pull request merged**